RealmSwift/Sync.swift

```diff
@@ -787,3 +787,166 @@ extension FunctionCallable {
     }
 }
 #endif // swift(>=5.5)
+
+public extension User {
+    // TODO: Flexible Sync - Add docstrings
+    func flexibleSyncConfiguration() -> Realm.Configuration {
+        let config = self.__flexibleSyncConfiguration()
+        return ObjectiveCSupport.convert(object: config)
+    }
+}
+
+// FLEXIBLE SYNC
+public protocol AnySubscription {
+}
+
+public class Subscription<Element: Object>: AnySubscription {
+    public typealias QueryFunction = (Query<Element>) -> Query<Element>
+
+    // When the subscription was created. Recorded automatically.
+    public var createdAt: Date = Date()
+
+    // When the subscription was last updated. Recorded automatically.
+    public var updatedAt: Date = Date()
+
+    // Name of the subscription, if not specified it will return the value in Query
+    public var name: String = ""
+
+    // Update query for subscription
+    public func update(@QueryBuilder _ to: () -> (AnySubscription)) throws {
+        fatalError()
+    }
+
+    private(set) public var query: QueryFunction
+    public init(name: String = "", query: @escaping QueryFunction) {
+        self.name = name
+        self.query = query
+    }
+}
+
+public protocol QueryBuilderComponent {}
+
+@resultBuilder public struct QueryBuilder {
+    public static func buildBlock(_ components: AnySubscription...) -> [AnySubscription] {
+        return components
+    }
+
+    public static func buildBlock(_ component: AnySubscription) -> AnySubscription {
+        return component
+    }
+
+}
+
+public protocol AnyQueryBuilderComponent {}
+
+// Realm operations
+// Realm will only allow getting all the subscriptions and subscribe to a query
+extension Realm {
+    // Get all subscriptions for this Realm.
+    /*private(set)*/ public var subscriptions: [AnySubscription] {
+        fatalError()
+    }
+}
+
+// TODO: Can we observer changes on the subscription set?
+// Task to get state changes from the write transaction
+@frozen public struct SubscriptionTask {
```

Comment:
This will allow us do this, we can think in another approach for this later
       ```
 try subscriptions.write {
            subscriptions.append {
                SyncSubscription<Contact>(name: "contacts-ny") {
                    $0.address.state == "NY" && $0.age > 10
                }
            }
        }
        .observe { state in
            if case .complete = state {

            }
        }
```